src/main/scala/controllers/ButtonAction.scala

```diff
@@ -8,12 +8,15 @@ case object Save extends ButtonAction
 
 case object Preview extends ButtonAction
 
+case object SaveItem extends ButtonAction
+
 object ButtonAction {
   def unapply(s: String): Option[ButtonAction] = {
     s match {
       case "_complete_button" => Some(Complete)
```

Comment:
The unapply is designed to ignore a mis-match, hence the Option[] return type - allows it to be used in pattern matching. But the way we're using it is to pass all of the url-encoded form values to it and see if any match an action.

The only thing with the hard-coded strings is that they need to match the values used on the form, so having a reverse mapper of some sort would make sense.